substrate/client/transaction-pool/src/fork_aware_txpool/view_store.rs

```diff
@@ -110,6 +110,43 @@ where
 		HashMap::<_, _>::from_iter(results.into_iter())
 	}
 
+	/// Synchronously imports single unverified extrinsics into every active view.
+	pub(super) fn submit_local(
+		&self,
+		xt: ExtrinsicFor<ChainApi>,
+	) -> Result<ExtrinsicHash<ChainApi>, ChainApi::Error> {
+		let active_views = {
+			self.active_views
+				.read()
+				.iter()
+				.map(|(_, view)| view.clone())
+				.collect::<Vec<_>>()
+		};
+
+		let tx_hash = self.api.hash_and_length(&xt).0;
+
+		let result = active_views
+			.iter()
+			.map(|view| {
+				self.dropped_stream_controller
+					.add_initial_views(std::iter::once(tx_hash), view.at.hash);
+				view.submit_local(xt.clone())
+			})
+			.reduce(|mut r, v| {
+				if r.is_err() && v.is_ok() {
+					r = v;
+				}
+				r
+			});
+
```

Comment:
Nit: Since we are using itertools anyway, this should be the same and a bit easier to understand.
```suggestion
			.find_or_first(|r| r.is_ok());

```

Reply:
added `find_or_first(Result::is_ok)` in 429b6f694b2